test/removed-dub-obj.sh

```diff
@@ -3,19 +3,13 @@
 . $(dirname "${BASH_SOURCE[0]}")/common.sh
 cd ${CURR_DIR}/removed-dub-obj
 
-DUB_CACHE_PATH="$HOME/.dub/cache/removed-dub-obj/"
+DUB_CACHE_PATH="$HOME/.dub/cache/removed-dub-obj"
 
-rm -rf $DUB_CACHE_PATH
+rm -rf "$DUB_CACHE_PATH"
 
 ${DUB} build --compiler=${DC}
 
-[ -d "$DUB_CACHE_PATH/obj" ] && die $LINENO "$DUB_CACHE_PATH/obj was found"
+[ -d "$DUB_CACHE_PATH" ] || die $LINENO "$DUB_CACHE_PATH not found"
 
-if [[ ${DC} == *"ldc"* ]]; then
-    if [ ! -f $DUB_CACHE_PATH/~master/build/library-*/obj/test.o* ]; then
-        ls -lR $DUB_CACHE_PATH
-        die $LINENO '$DUB_CACHE_PATH/~master/build/library-*/obj/test.o* was not found'
-    fi
-fi
-
-exit 0
+numObjectFiles=$(find "$DUB_CACHE_PATH" -type f -iname '*.o*' | wc -l)
+[ "$numObjectFiles" -eq 0 ] || die $LINENO "Found left-over object files in $DUB_CACHE_PATH"
```

Comment:
this change seems to revert what has previously been published in the changelogs: https://github.com/kinke/dub/commit/176ea9cc13c9c36f285dea628954d049a04aa0f2

> Enables co-existence and parallel compilation of the same project with different settings (e.g. cross compilation) by moving `.dub/obj` to `$DUB_TARGET_PATH/obj`.

Although that changelog entry seems faulty right now, since it didn't actually write to the target path (e.g. `"targetPath": "bin"`) but to the build cache path.

So I think this change is fine.

Reply:
With LDC v1.25+, this PR solves the concurrency problem more robustly - each compiler invocation uses a unique temp dir for its object files, no need to specify unique-ish paths explicitly.